pennylane/ops/op_math/__init__.py

```diff
@@ -124,6 +135,12 @@
     "ControlledQubitUnitary",
     "CY",
     "CZ",
+    "CNOT",
```

Comment:
I don't know if you've already taken this into account, but what I would ideally like to see is this behaviour:
> - qml.ctrl will flatten nested controlled operators to a single multi-controlled operation.

be applied not only during first construction of the operator, but also in later stages. Consider an operator:
```py
class MyOp:
  def decomposition(self):
    return [qml.CNOT(self.wires)]
```

If the user writes `qml.ctrl(MyOp([0,1]), control=[1,2])`, the construction logic cannot initially generate a `MultiControlledX` gate. However, it would be great to ensure that during the decomposition process, when `MyOp` turns into `CNOT`, we generate a `MultiControlledX` rather than a `C(CNOT)`.

Reply:
> However, it would be great to ensure that during the decomposition process, when MyOp turns into CNOT, we generate a MultiControlledX rather than a C(CNOT).

I believe this is indeed the new behaviour, but I can add tests cases to make sure.